nmdc_server/api.py

```diff
@@ -862,22 +865,63 @@ async def delete_submission(
     return Response(status_code=status.HTTP_204_NO_CONTENT)
 
 
+@router.put("/metadata_submission/{id}/lock")
+async def lock_submission(
+    response: Response,
+    id: str,
+    db: Session = Depends(get_db),
+    user: models.User = Depends(get_current_user),
+) -> schemas.LockOperationResult:
+    submission: Optional[SubmissionMetadata] = db.query(SubmissionMetadata).get(id)
+    if not submission:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Submission not found")
```

Comment:
Should we prevent users who don't have any access (i.e. `crud.can_read_submission(...)`) from being able to obtain a lock on a submission?

Reply:
Maybe this should be done in `crud.try_get_submission_lock` itself now that I'm thinking about it. In either case, this wasn't being done before so if you want to defer it/open up an issue to track this separately that's fine by me.

Reply:
Let me take a look at how much work / how much impact it would be to do that now.

Reply:
I added a role check to `try_get_submission_lock`. Seems like a good safeguard to have in place.